readthedocs/projects/views/public.py

```diff
@@ -299,7 +300,7 @@ def verify_project_token(cls, token, project_slug):
 project_badge = never_cache(ProjectBadgeView.as_view())
 
 
-class ProjectDownloadMediaBase(CDNCacheControlMixin, ServeDocsMixin, View):
+class ProjectDownloadMediaBase(CDNCacheControlMixin, CDNCacheTagsMixin, ServeDocsMixin, View):
```

Comment:
We don't need to do any changes to the mixin or upstream logic, we just need to add the `_get_project` and `_get_version` methods to this VIew, and all the existing logic should just work.

Reply:
I still prefer to keep using the current pattern defining these two methods and refactor this as we want in another PR. Consistency is more important to me until we do the refactor and change it to a new implementation.

Reply:
Not sure we get any benefit from being conservative here on code changes. But you can review just those changes here:

- #12503 

I'm retaining this PR as the refactor PR.

Reply:
I think we don't benefit from being pedantic, but doing half refactors and keeping both patterns in the codebase isn't a positive step forward, but makes things more confusing. So normally we try to follow the existing pattern until we can properly refactor.